[AI Tools] - S3 compliance (#4609)
ovh/docs: pages/platform/ai/gi_08_s3_compliance/guide.en-gb.md

---
title: Data - S3 compliance with AI Tools
slug: s3-compliance
excerpt: Learn how to use S3 buckets with AI Tools
section: General information
order: 110
updated: 2023-05-10
---

**Last updated 10th May, 2023.**

## Objective

This guide is intended to help you to use `S3 buckets` with **AI Tools** through OVHcloud Control Panel and `ovhai` CLI. This will help you to manage your data when working with AI Notebooks, AI Training and AI Deploy.

## Requirements

- A **Public Cloud project**
- Access to the [OVHcloud Control Panel](https://www.ovh.com/auth/?action=gotomanager&from=https://www.ovh.co.uk/&ovhSubsidiary=GB)
- A working `ovhai` CLI. See our guide on [how to install ovhai CLI](https://docs.ovh.com/gb/en/publiccloud/ai/cli/install-client).

## Instructions

In the next steps, you will be able to **create a user to manage your S3 buckets**, **add a new datastore** dedicated to your buckets, then **create and increment a bucket**.

> Then, you will be able to use this bucket with the different **OVHcloud AI Tools** through the `ovhai` CLI.

Reviewer: And also through the OVHcloud control panel ?

>

You have here the basic command to create an OVHcloud S3 datastore:

```console
ovhai datastore add s3 <alias> https://s3.<region>.io.cloud.ovh.net/ <region> <my-access-key> <my-secret-key>
```

In this example, the datastore created will be located in Gravelines (`GRA`) and its alias will be `S3GRA`.

```console
ovhai datastore add s3 S3GRA https://s3.gra.io.cloud.ovh.net/ gra <my-access-key> <my-secret-key> --store-credentials-locally
```

You can now check the datastore list:

```console
ovhai datastore list
```

You should see your S3 datastore in the list.

```console
ALIAS STORE_TYPE OWNER    ENDPOINT
BHS   swift      ovhcloud ~
DE    swift      ovhcloud ~
GRA   swift      ovhcloud ~
S3GRA s3         customer https://s3.gra.io.cloud.ovh.net/
SBG   swift      ovhcloud ~
SGP   swift      ovhcloud ~
UK    swift      ovhcloud ~
WAW   swift      ovhcloud ~
```

Now that you have a dedicated datastore, you can create some S3 buckets through OVHcloud Control Panel and `ovhai` CLI.

### Create an S3 bucket

You have two ways to [create a new S3 bucket](/pages/cloud/storage/object_storage/s3_create_bucket). The first one is to use the UI.

#### Using the Control Panel

> To create an **Object Storage bucket**, first log in to your [OVHcloud Control Panel](https://www.ovh.com/auth/?action=gotomanager\&from=https://www.ovh.co.uk/\&ovhSubsidiary=GB) and open your `Public Cloud` project. Click on `Object Storage` in the side bar and then on the `My containers` tab. Now, you can click on `Create Object Container`.

Reviewer: ```suggestion
To create an **Object Storage bucket**, first log in to your [OVHcloud Control Panel](https://www.ovh.com/auth/?action=gotomanager\&from=https://www.ovh.co.uk/\&ovhSubsidiary=GB) and open your `Public Cloud` project. Click on `Object Storage` in the side bar and then on the `My containers` tab. Now, click the `Create an object container` button.
```
